### PR TITLE
test(api): overhaul API tests

### PR DESCRIPTION
**Pull request by b5**
our API tests are brittle & coupled up in a few mega-functions. Time to break those apart &
provide an easy update mecahnism for when breaking API changes are intentional.
I've *heavily* refactored the api tests, working individual handlers & comparing responses
to abide records instead of json files.

### CODE REVIEW

File: api/api_test.go [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+	// we use the dsfs.Timestamp func variable to override
+	// the actual time
+	prevTs := dsfs.Timestamp
+	dsfs.Timestamp = func() time.Time { return time.Date(2001, 01, 01, 01, 01, 01, 01, time.UTC) }
+
+	if r, err = test.NewTestRepo(rc); err != nil {
+		t.Fatalf("error allocating test repo: %s", err.Error())
+	}
+
+	lib.Config = config.DefaultConfig()
```
Comment (dustmop) [inline api/api_test.go]: Please use DefaultConfigForTesting from config/test_config.go : https://github.com/qri-io/qri/blob/31be72d4e30742ac7ea731600f66f769ead3afb1/config/test_config.go#L8

This avoids generating private keys, using pregenerated keys instead.

File: api/api_test.go [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+
+	client := &http.Client{}
+
+	r, err := test.NewTestRepo(nil)
+	if err != nil {
+		t.Errorf("error allocating test repo: %s", err.Error())
+		return
+	}
+
+	s, err := New(r, func(opt *config.Config) {
```
Comment (dustmop) [inline api/api_test.go]: This will conflict with #536, btw.

File: api/peers_test.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+	"github.com/qri-io/qri/config"
+	"github.com/qri-io/qri/p2p"
+)
+
+func TestPeerHandlers(t *testing.T) {
+	r, teardown := newTestRepo(t)
+	defer teardown()
+
+	node, err := p2p.NewQriNode(r, func(o *config.P2P) {
```
Comment (dustmop) [inline api/peers_test.go]: Another conflict

File: api/api_test.go [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+	defer teardown()
+
+	h := NewRootHandler(NewDatasetHandlers(r, false), NewPeerHandlers(r, nil, false))
+	rootCases := []handlerTestCase{
+		{"OPTIONS", "/", nil},
+		{"GET", "/", nil},
+	}
+	runHandlerTestCases(t, "root", h.Handler, rootCases)
+
+	healthCheckCases := []handlerTestCase{
```
Comment (dustmop) [inline api/api_test.go]: These two arrays are the same, will they always be the same? How about just using the same one twice.

File: api/render_test.go [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+import (
+	"testing"
+)
+
+func TestRenderHandler(t *testing.T) {
+	r, teardown := newTestRepo(t)
+	defer teardown()
+
+	cases := []handlerTestCase{
```
Comment (dustmop) [inline api/render_test.go]: Please add a comment why this is disabled.

File: api/api_test.go (hunk 4 above)
Comment (b5) [reply api/api_test.go]: They should diverge over time, just wanted to get them in place as such for basic coverage.

**Comment by b5** [comment]
LGTM emoji-acronym courtesy of @ramfox

**Pull request merged**